### DISCUSSION TITLE
Help - reportimgs.js dosn't contain all subjects

### DISCUSSION THREAD

**Discussion by kriecho** [HALFpipe/HALFpipe#327] [Q&A] [answered]
Hello,

I successfully preprocessed 1400 subjects with Halfpipe 1.2.1. However, I have a problem with the reports. When opening index.html only ~250 subjects are shown and my reportimgs.js also only contains those ~250 subjects. However, 1400 subjects are listed in reportpreproc.js as "done" as well as all 1400 subjects are having values in reportvals.js. 

I really would appreciate help, how I can repair the reportimgs.js without rerunning all subjects.

Thanks in advance

**Comment by HippocampusGirl** [comment]
Dear @kriecho,

This seems like a very useful feature to add to `HALFpipe`, but it would come with a bit of development effort and I don't think I can allocate time for it right now. But maybe we can find an easier solution :-)

- Were there any error messages related to the `.js` files?

- How many `.svg` files exist in the subject-specific folders of the `reports` directory? Are all the images that should be there according to the manual actually there?

Best,
Lea

**Comment by kriecho** [comment, marked as answer]
Hi Lea, 

thanks for the quick reply.
- The report folder contained 3 .js files ( reportimgs.js, reportpreproc.js, reportvals.js). 
- Regarding the .svg files, it seems that each subject is having 12 files. 
- I did not find any error messages in the error-log-files related to the .js files.

However, I also tried to recreate the reportimgs.js file meanwhile with Matlab, but JS is quite tricky/stubborn... 

Best,
Elischa

**Reply by HippocampusGirl** [reply]
Hi Elischa,

That means all the files are there, we just need to create the `reportimgs.js` file. That makes it a lot easier!

I think your MATLAB script looks very good, it just needs to have a different hash for each image. There are not much requirements for this, for example, you could just number them or [generate 32 random hexadecimal characters](https://de.mathworks.com/matlabcentral/answers/614558-how-do-i-create-random-hex-characters) for each image.

Best,
Lea

**Reply by kriecho** [reply]
Hi Lea, 
thanks again for the fast response. The hash values were driving me crazy as I tried recreate them from the file... 🙈 
Now it seems, as the script for recreating the reportimgs.js file is working.
Best,
Elischa
Free to use Matlab-Script for recreating reportimgs.js: [recreate_reportimgs_halfpipe.m.txt](https://github.com/HALFpipe/HALFpipe/files/8716518/recreate_reportimgs_halfpipe.m.txt)

**Reply by HippocampusGirl** [reply]
Hi Elischa, 

That's awesome :-)

Best,
Lea
